### PR TITLE
feat: :sparkles: species show/create page

### PR DESCRIPTION
**Pull request by harshzalavadiya**
### Species Create
  - [x] validate
    - [x] full match
    - [x] partial match
    - [x] manual
      - [x] validate

### Species Show

- [ ] related document 🔨
- [ ] groups (partial) 🔨
- [ ] payload cleanup
- [ ] mail
- [x] preferred common name API + Species Icon
- [x] follow/unfollow
- [x] delete species
- [x] breadcrumbs
- [x] gallery
- [x] taxon browser
- [x] synonym edit
- [x] common name edit
- [x] temporal destribution
- [x] related observations
- [x] species fields
- [x] create
- [x] edit
  - [x] references show
  - [x] edit (incl. references)
- [x] delete
- [x] resources
- [x] trait
- [x] occurance map
- [x] references
- [x] combined
- [x] assigned to page
- [x] permissions (partial)
- [x] activity

### Bugfixes
- [x] file picker z-index
- [x] field title field prevent add/edit
- [x] references add glitch `key`

```txt
* 🔨 API Pending
* 🎨 UI Pending
```
---

### CODE REVIEW

File: src/components/pages/species/show/fields/field/index.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+      (currentField.values.length || currentField.traits.length || childFieldsHasValue) &&
+      valueCallback
+    ) {
+      setChildFieldsHasValue(true);
+      valueCallback(true);
+    }
+  }, [currentField, childFieldsHasValue]);
+
+  // References
+  if (parentField.id === 81) {
```
Comment (teamdraftbox) [inline src/components/pages/species/show/fields/field/index.tsx]: can below 3  "if" line be put in switch case format

File: src/components/pages/species/list/header/index.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { PageHeading } from "@components/@core/layout";
+import useTranslation from "@hooks/use-translation";
+import React from "react";
+
+export default function Header() {
```
Comment (teamdraftbox) [inline src/components/pages/species/list/header/index.tsx]: if this component is not shared ,could pls use it directly in its desired component itself

**Review by teamdraftbox — APPROVED** [review]
i have  requested for some minor changes, apart from that i have done a overall riview it looks good to me, please go ahead and merge it

File: src/components/pages/species/show/fields/field/index.tsx (hunk 1 above)
Comment (harshzalavadiya) [reply src/components/pages/species/show/fields/field/index.tsx]: @teamdraftbox done!

File: src/components/pages/species/list/header/index.tsx (hunk 2 above)
Comment (harshzalavadiya) [reply src/components/pages/species/list/header/index.tsx]: @teamdraftbox done!

**Pull request merged**